src/cls/rbd/cls_rbd.cc

```diff
@@ -4423,6 +4518,65 @@ int mirror_instances_remove(cls_method_context_t hctx, bufferlist *in,
 }
 
 /**
+ * Input:
+ * @param start_after: key to start after
+ * @param max_return: max return items
+ *
+ * Output:
+ * @param std::map<std::string, cls::rbd::MirrorImageMap>: image map
+ * @returns 0 on success, negative error code on failure
+ */
+int mirror_image_map_list(cls_method_context_t hctx, bufferlist *in,
+                          bufferlist *out) {
+  std::string start_after;
+  uint64_t max_return;
+  try {
+    bufferlist::iterator it = in->begin();
+    ::decode(start_after, it);
+    ::decode(max_return, it);
+  } catch (const buffer::error& err) {
+    return -EINVAL;
+  }
+
+  std::map<std::string, cls::rbd::MirrorImageMap> image_map;
+  int r = mirror::mirror_image_map_list(
+    hctx, start_after, max_return, &image_map);
+  if (r < 0) {
+    return r;
+  }
+
+  ::encode(image_map, *out);
+  return 0;
+}
+
+/**
+ * Input:
+ * @param global_id: image global id
+ * @param cls::rbd::MirrorImageMap: image map
+ *
+ * Output:
+ * @returns 0 on success, negative error code on failure
+ */
+int mirror_image_map_update(cls_method_context_t hctx, bufferlist *in,
+                            bufferlist *out) {
+  std::map<std::string, std::string> map_update;
```

Comment:
Revert to the original version of this method (and ```mirror_image_map_remove```) -- multiple updates can be packed into a single librados operation.  Definitely do not just pass a string for the data since that means we could never improve it (i.e. add stats for load balancing)